### PR TITLE
cortx-32490: Initial commit for sns_direct_rebalance

### PR DESCRIPTION
**Pull request by mehjoshi**
Signed-off-by: Mehul Joshi <mehul.joshi@seagate.com>

# Problem Statement
- This is the initial commit for SNS direct rebalance. It adds code for fop/fom creation and other basic changes for direct rebalance.

# Design
-  The design document is not ready as yet but the same will be provided prior to the merge from the feature branch

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: ha/halon/interface.c [hunk 1]
```diff
@@ -770,6 +770,7 @@ static int halon_interface_level_enter(struct m0_module *module)
 	case M0_HALON_INTERFACE_LEVEL_SNS_CM_TRIGGER_FOPS:
 		m0_sns_cm_repair_trigger_fop_init();
 		m0_sns_cm_rebalance_trigger_fop_init();
+		m0_sns_cm_dtrebalance_trigger_fop_init();
```
Comment (nkommuri) [inline ha/halon/interface.c:773]: please add corresponding ...fop_fini() in halon_interface_level_leave() function

File: scripts/m0 [hunk 2]
```diff
@@ -88,11 +88,11 @@ _make() {
 
     if [ "${TARGET:-}" = rpms ]; then
         ## Tolerate warnings. rpms get built in spite of them.
-        make ${MAKE_OPTS:-} $TARGET || exit $?
+        bear make ${MAKE_OPTS:-} $TARGET || exit $?
```
Comment (nkommuri) [inline scripts/m0:91]: why "bear?" Is "bear" installed by default?

File: sns/cm/trigger_fop_common.c [hunk 3]
```diff
@@ -67,7 +69,9 @@ M0_INTERNAL int m0_sns_cm_trigger_fop_alloc(struct m0_rpc_machine  *mach,
 		[CM_OP_REPAIR_STATUS]    = &m0_sns_repair_status_fopt,
 		[CM_OP_REBALANCE_STATUS] = &m0_sns_rebalance_status_fopt,
 		[CM_OP_REPAIR_ABORT]     = &m0_sns_repair_abort_fopt,
-		[CM_OP_REBALANCE_ABORT]  = &m0_sns_rebalance_abort_fopt
+		[CM_OP_REBALANCE_ABORT]  = &m0_sns_rebalance_abort_fopt,
+		[CM_OP_REBALANCE_ABORT]  = &m0_sns_rebalance_abort_fopt,
```
Comment (nkommuri) [inline sns/cm/trigger_fop_common.c:73]: Duplicate entry

**Review by nkommuri — APPROVED** [review]

File: scripts/m0 (hunk 2 above)
Comment (mehjoshi) [reply scripts/m0:91]: sorry, this change was pushed accidentally. I will remove it. Thanks!

File: sns/cm/trigger_fop_common.c (hunk 3 above)
Comment (mehjoshi) [reply sns/cm/trigger_fop_common.c:73]: Done

File: ha/halon/interface.c (hunk 1 above)
Comment (mehjoshi) [reply ha/halon/interface.c:773]: Done.

**Pull request merged**